lib/src/signed_video_tlv.c

```diff
@@ -900,6 +902,40 @@ tlv_find_tag(signed_video_t *self,
   return NULL;
 }
 
+svi_rc
+tlv_find_tag_and_decode(signed_video_t *self,
+    const uint8_t *tlv_data,
+    size_t tlv_data_size,
+    sv_tlv_tag_t tag,
+    bool with_ep)
+{
+  svi_rc status = SVI_UNKNOWN;
+
+  if (!self || !tlv_data || tlv_data_size == 0) return SVI_INVALID_PARAMETER;
+
+  const uint8_t *tag_ptr = tlv_find_tag(self, tlv_data, tlv_data_size, tag, with_ep);
+
+  if (!tag_ptr) {
+    return SVI_INVALID_PARAMETER;
+  }
+
+  // Read the length
```

Comment:
Suggestion: Instead of reading the tag length and then use tlv_decode() with that length (to tell it we only want to read the first tag), I think it is more clear to simply decode_tlv_header(), get_decoder() and then decode()

Something like this, with proper error handling?
```
size_t tlv_header_size = 0;
size_t length = 0;
status = decode_tlv_header(tlv_data, &tlv_header_size, &tag, &length);
sv_tlv_decoder_t decoder = get_decoder(tag);
status = decoder(self, tlv_data + tlv_header_size, length);
```